src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -443,9 +443,21 @@ public MessageBuilder appendFailedTests() {
             return this;
         }
 
-        public MessageBuilder appendCustomMessage() {
+        public MessageBuilder appendCustomMessage(Result buildResult) {
             String customMessage = notifier.getCustomMessage();
-
+            if (buildResult != null) {
```

Comment:
Thanks @KreAch3R 
This looks great

One problem though imo is that this is a breaking change.

Currently when someone uses the custom message it applies to all statuses

With this change it only applies to the `Notify Build Start`

I would suggest something like:
`Custom Message - All Statuses` - Matches the existing field

`Custom Message - Build Start` - new field if someone only wants to override the build start message

Reply:
Thank you @timja 
You are right, it was a breaking change.

Check the latest commit I pushed. I think this restores the default behavior. The custom message will be applied to every build result, except if there is a specific custom message for that result.

Reply:
What if a user wants a start message but none of the others?

Reply:
I think I'm gonna leave it like that for this commit (so the default behavior remains the same, the custom message is applied everywhere). Your idea about the custom 'build start' and 'all statuses' is good though, I'll see if i can work on it in the future